### ISSUE TITLE
GQLExecutor memory leak

### ISSUE THREAD

**Issue by Dschoordsch** [ParabolInc/parabol#9262]
The GQLExecutor memory consumption goes steadily up. As seen on [here](https://app.datadoghq.com/dashboard/xjt-8zg-if3/parabol-summary?fullscreen_end_ts=1701145224147&fullscreen_paused=true&fullscreen_refresh_mode=paused&fullscreen_section=overview&fullscreen_start_ts=1700576749088&fullscreen_widget=4&refresh_mode=sliding&tpl_var_project_id%5B0%5D=parabol-saas-staging&from_ts=1701083197023&to_ts=1701687997023&live=true) 🔒(top three lines)
<img width="1318" alt="Screenshot 2023-12-04 at 12 13 14" src="https://github.com/ParabolInc/parabol/assets/7331043/3476d385-1ead-4d0a-b138-f710e7dcff4e">
the memory usage stayed high even through the weekend where I would expect it to drop.

**Comment by Dschoordsch** [comment]
Labelling it P3 as it's not critical now. Could be related to #8414

**Comment by Dschoordsch** [comment]
@rafaelromcar-parabol This could be an issue if we don't release for some time, like over christmas

**Comment by rafaelromcar-parabol** [comment]
We should keep an eye on it, for sure. But keep in mind that if the containers arrive to 4 GB of memory, they will be automatically restarted.

That said, we could restart the application next week before Christmas if needed. During the week or on the weekend. It can be done easily using the last pipeline in the Gitlab repository.

**Comment by rafaelromcar-parabol** [comment]
Also, I think this should be a P2. Mem leaks should always be fixed asap as they can affect stability. Even more in our application, that doesn't like restarting components very much. At least it didn't before [this](https://github.com/ParabolInc/parabol/pull/9252). Let's see after the release this week.

**Comment by mattkrick** [comment]
Blocked behind #9282

**Comment by rafaelromcar-parabol** [comment]
I was checking the memory usage on the GQL Ex and they are fine but at more than 60%. Which is fine, because if we do not release, the worst that can happen is having Kubernetes restarting them if they reach 100%. It isn't a big problem. Just sharing my thoughts here.

**Comment by mattkrick** [comment]
true for the saas! for certain clients who do do not manage their infra as diligently as we do, restarts may pose a problem 😅

**Comment by rafaelromcar-parabol** [comment]
Of course! I was talking about the Christmas period for the SaaS 😸 PPMIs haven't been updated in a long while. They are still in 6.83.0 and they have been like that for a while.

**Comment by jordanh** [comment]
Prioritized to backlog

**Comment by jordanh** [comment]
After moving to yoga, we can now close this.
